### PR TITLE
Added some error handling

### PR DESCRIPTION
**Pull request by gabrielgradinaru**
#4 

-Added timeout when making request http connection
-Added try/except in case a proxy parser fails

### CODE REVIEW

**Comment by gabrielgradinaru** [comment]
Hi,

- I have added the configurable timeout on the UrlParser base class
- For the try/except I have added ReadTimeout exception for now but there could also be other exceptions in the future and will have to be added
- I think the best approach for Class TestProxyProviders is to mock the requests response because, you've said yourself that proxy websites are not reliable

File: http_request_randomizer/requests/proxy/requestProxy.py [hunk 1]
```diff
@@ -37,9 +37,9 @@ def __init__(self, web_proxy_list=[], sustain=False):
         # Each of the classes below implements a specific URL Parser
         #####
         parsers = list([])
-        parsers.append(FreeProxyParser('http://free-proxy-list.net'))
-        parsers.append(ProxyForEuParser('http://proxyfor.eu/geo.php', 1.0))
-        parsers.append(RebroWeeblyParser('http://rebro.weebly.com'))
+        parsers.append(FreeProxyParser('http://free-proxy-list.net', timeout=timeout))
+        parsers.append(ProxyForEuParser('http://proxyfor.eu/geo.php', 1.0, timeout=timeout))
+        parsers.append(RebroWeeblyParser('http://rebro.weebly.com', timeout=timeout))
         # parsers.append(SamairProxyParser('http://samair.ru/proxy/time-01.htm'))
```
Comment (pgaref) [inline http_request_randomizer/requests/proxy/requestProxy.py]: SemairProxy seems to be working now  - we should uncomment this line.
Also if timeout is working it won't cause issues in the future

File: http_request_randomizer/requests/proxy/requestProxy.py [hunk 2]
```diff
@@ -51,7 +51,10 @@ def __init__(self, web_proxy_list=[], sustain=False):
         self.parsers = parsers
         self.proxy_list = web_proxy_list
         for i in range(len(parsers)):
-            self.proxy_list += parsers[i].parse_proxyList()
+            try:
+                self.proxy_list += parsers[i].parse_proxyList()
+            except ReadTimeout:
+                self.logger.debug("Proxy Parser failed for: {}".format(parsers[i].url))
```
Comment (pgaref) [inline http_request_randomizer/requests/proxy/requestProxy.py]: Let's log a warning message instead - it should be something like: self.logger.warn("Proxy Parser: {} TimedOut!".format(parsers[i].url))

Comment (pgaref) [inline http_request_randomizer/requests/proxy/requestProxy.py]: These minor changes will do. Also lets mock the Requests in TestProxyProviders class as you suggested.
The changes need to be on the same PR as they tests currently fail!

Thanks :)

**Comment by pgaref** [comment]
@gabrielgradinaru great job! Thanks for contributing :)

**Pull request merged**